crates/solidity/outputs/cargo/crate/src/backend/ast/generated/mutator.rs

```diff
@@ -0,0 +1,2963 @@
+                UsingClause::UsingDeconstruction(
+                    self.mutate_using_deconstruction(using_deconstruction),
+                )
+            }
+        }
+    }
+    fn mutate_using_clause(&mut self, source: &UsingClause) -> UsingClause {
+        self.default_mutate_using_clause(source)
+    }
+
+    fn default_mutate_using_operator(&mut self, source: &UsingOperator) -> UsingOperator {
+        match source {
+            UsingOperator::Ampersand => UsingOperator::Ampersand,
+            UsingOperator::Asterisk => UsingOperator::Asterisk,
+            UsingOperator::BangEqual => UsingOperator::BangEqual,
+            UsingOperator::Bar => UsingOperator::Bar,
+            UsingOperator::Caret => UsingOperator::Caret,
+            UsingOperator::EqualEqual => UsingOperator::EqualEqual,
+            UsingOperator::GreaterThan => UsingOperator::GreaterThan,
+            UsingOperator::GreaterThanEqual => UsingOperator::GreaterThanEqual,
+            UsingOperator::LessThan => UsingOperator::LessThan,
+            UsingOperator::LessThanEqual => UsingOperator::LessThanEqual,
+            UsingOperator::Minus => UsingOperator::Minus,
+            UsingOperator::Percent => UsingOperator::Percent,
+            UsingOperator::Plus => UsingOperator::Plus,
+            UsingOperator::Slash => UsingOperator::Slash,
+            UsingOperator::Tilde => UsingOperator::Tilde,
+        }
+    }
+    fn mutate_using_operator(&mut self, source: &UsingOperator) -> UsingOperator {
+        self.default_mutate_using_operator(source)
+    }
+
+    fn default_mutate_using_target(&mut self, source: &UsingTarget) -> UsingTarget {
+        match source {
+            UsingTarget::TypeName(ref type_name) => {
+                UsingTarget::TypeName(self.mutate_type_name(type_name))
+            }
+            UsingTarget::Asterisk => UsingTarget::Asterisk,
+        }
+    }
+    fn mutate_using_target(&mut self, source: &UsingTarget) -> UsingTarget {
+        self.default_mutate_using_target(source)
+    }
+
+    fn default_mutate_contract_specifier(
+        &mut self,
+        source: &ContractSpecifier,
+    ) -> ContractSpecifier {
+        match source {
+            ContractSpecifier::InheritanceSpecifier(ref inheritance_specifier) => {
+                ContractSpecifier::InheritanceSpecifier(
+                    self.mutate_inheritance_specifier(inheritance_specifier),
+                )
+            }
+            ContractSpecifier::StorageLayoutSpecifier(ref storage_layout_specifier) => {
+                ContractSpecifier::StorageLayoutSpecifier(
+                    self.mutate_storage_layout_specifier(storage_layout_specifier),
+                )
+            }
+        }
+    }
+    fn mutate_contract_specifier(&mut self, source: &ContractSpecifier) -> ContractSpecifier {
+        self.default_mutate_contract_specifier(source)
+    }
+
+    fn default_mutate_contract_member(&mut self, source: &ContractMember) -> ContractMember {
```

Comment:
this + other traits:

WDYT of moving the default methods outside the trait?
to prevent the user from accidentally overriding them in their implementation...
i.e. something like this:

```rust
struct DefaultMutator;

impl DefaultMutator {
    fn mutate_foo(mutator: &mut impl Mutator, source: &Foo) -> Foo {
        // ....
    }
}
```